clients/database/migrations/20230523101344_partition_proxied_request_metrics_table.up.sql

```diff
@@ -0,0 +1,94 @@
+    CREATE TABLE proxied_request_metrics_year2023month5_day19 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-19 00:0:0.0') TO ('2023-05-19 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day20 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-20 00:0:0.0') TO ('2023-05-20 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day21 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-21 00:0:0.0') TO ('2023-05-21 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day22 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-22 00:0:0.0') TO ('2023-05-22 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day23 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-23 00:0:0.0') TO ('2023-05-23 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day24 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-24 00:0:0.0') TO ('2023-05-24 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day25 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-25 00:0:0.0') TO ('2023-05-25 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day26 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-26 00:0:0.0') TO ('2023-05-26 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day27 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-27 00:0:0.0') TO ('2023-05-27 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day28 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-28 00:0:0.0') TO ('2023-05-28 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day29 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-29 00:0:0.0') TO ('2023-05-29 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day30 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-30 00:0:0.0') TO ('2023-05-30 23:59:59.999999');
+    CREATE TABLE proxied_request_metrics_year2023month5_day31 PARTITION OF proxied_request_metrics
+    FOR VALUES FROM ('2023-05-31 00:0:0.0') TO ('2023-05-31 23:59:59.999999');
+-- new partitions will be auto-created by the metric compaction routines
+
+-- copy the data to our new partitioned table
+INSERT INTO proxied_request_metrics (method_name,block_number,response_latency_milliseconds,request_time,request_ip,hostname,user_agent,referer,origin) SELECT method_name,block_number,response_latency_milliseconds,request_time,request_ip,hostname,user_agent,referer,origin FROM proxied_request_metrics_old;
+
+-- get rid of the old table
+DROP TABLE proxied_request_metrics_old;
+
+-- set up previous indices
+CREATE INDEX  IF NOT EXISTS method_name_idx ON proxied_request_metrics(method_name);
```

Comment:
yeah, because doing it first results in the indices not being created (thats the why I originally did it and not sure why but doing it this way makes them get set)

Reply:
(you can try yourself locally by cloning this branch, switching the order, running `make` to build and start all the servcies, and then once the service is started running `make debug-database` and `\d proxied_request_metrics` to inspect the table